src/webgpu/gpu_test.ts

```diff
@@ -846,13 +854,26 @@ export class GPUTestBase extends Fixture<GPUTestSubcaseBatchState> {
       ${textureSrcCode}
       @group(0) @binding(1) var<storage, read_write> dst : Buffer;
 
-      @compute @workgroup_size(1) fn main() {
-        var coord = vec2<i32>(0, 0);
-        for (var sampleIndex = 0; sampleIndex < ${sampleCount};
+      struct Params {
+        origin: vec3u,
+        pad0: u32,
+        extent: vec3u,
+        pad1: u32,
+      };
+      @group(0) @binding(2) var<uniform> params : Params;
+
+      @compute @workgroup_size(${kWorkgroupSizeX}, ${kWorkgroupSizeY}, 1) fn main(@builtin(global_invocation_id) id : vec3u) {
+        let boundary = params.origin + params.extent;
+        let coord = params.origin + id;
+        if (any(coord >= boundary)) {
+          return;
+        }
+        let offset = (coord.x + coord.y * params.extent.x) * ${componentCount} * ${sampleCount};
```

Comment:
If I'm reading this correctly then the origin of the texture (0,0) would map to the beginning of the buffer [0]. But I think copyTextureToBuffer would map the origin of the _copy_ to the beginning of the buffer which is more practical so it doesn't require a ton of unused padding at the start of the buffer.

Reply:
You are right. Changed from using `coord.xy` to `id.xy`